### PR TITLE
Fixes missing files when manually building vscode server

### LINKED ISSUE #137568: missing files when manually building vscode server
Steps to Reproduce:

1. run `yarn gulp vscode-reh-web-linux-x64-min`
2. :bug: open output folder `vscode-reh-web-linux-x64/extensions/markdown-language-features` and observe `notebook-out` folder is missing, same for `markdown-math`

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes #137568

### CODE REVIEW

**Review by alexdima — APPROVED** [review]
@mjbvz @joaomoreno The suggested change looks good to me, but I would like to get approval from you as well. Here is what I can put together:
* `yarn extensions-ci` is executed in the compilation phase
* `extensions-ci` is a gulp task that executes both `compileExtensionsBuildTask` and `compileExtensionMediaBuildTask`
* in `gulpfile.vscode.reh.js`, we would be calling in 1 place `compileExtensionsBuildTask` but not `compileExtensionMediaBuildTask`
* similarly, in `gulpfile.vscode.js` we would be calling in 3 places `compileExtensionsBuildTask`, but not `compileExtensionMediaBuildTask`
* this PR adds `compileExtensionMediaBuildTask` in 2 of the 4 places where `compileExtensionsBuildTask` was called

I plan that `gulpfile.vscode.reh.js` should do exactly what `gulpfile.vscode.js` does. So in this particular case, although issue #137568 talks about missing files in the server build, it looks like those files would be missing also in the vscode build (unless someone manually runs `extensions-ci` before).

I will therefore defer to you @joaomoreno @mjbvz to figure out what you want to do here and I am happy to take whatever you do also in `gulpfile.reh.js`.

**Review by mjbvz — APPROVED** [review]

**Comment by joaomoreno** [comment]
Thanks for the investigation @alexdima. This is the correct change. The changes in `gulpfile.vscode.js` should've been made at the time `compileExtensionMediaBuildTask` was introduced.

The `*-ci` tasks are running only in our CI which is orchestrated in a way that `extensions-ci` is always going to be called before `vscode-*-ci`.

**Review by joaomoreno — APPROVED** [review]

**Pull request merged**
